controllers/components/workbenches/workbenches.go

```diff
@@ -0,0 +1,73 @@
+package workbenches
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	cr "github.com/opendatahub-io/opendatahub-operator/v2/pkg/componentsregistry"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/annotations"
+)
+
+type componentHandler struct{}
+
+func init() { //nolint:gochecknoinits
+	cr.Add(&componentHandler{})
+}
+
+func (s *componentHandler) GetName() string {
+	return componentsv1.WorkbenchesComponentName
+}
+
+func (s *componentHandler) GetManagementState(dsc *dscv1.DataScienceCluster) operatorv1.ManagementState {
+	return dsc.Spec.Components.Workbenches.ManagementState
+}
+
+func (s *componentHandler) NewCRObject(dsc *dscv1.DataScienceCluster) client.Object {
+	workbenchesAnnotations := make(map[string]string)
+
+	switch dsc.Spec.Components.Workbenches.ManagementState {
+	case operatorv1.Managed, operatorv1.Removed:
+		workbenchesAnnotations[annotations.ManagementStateAnnotation] = string(dsc.Spec.Components.Workbenches.ManagementState)
+	default: // Force and Unmanaged case for unknown values, we do not support these yet
+		workbenchesAnnotations[annotations.ManagementStateAnnotation] = "Unknown"
+	}
+
+	return client.Object(&componentsv1.Workbenches{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       componentsv1.WorkbenchesKind,
+			APIVersion: componentsv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        componentsv1.WorkbenchesInstanceName,
+			Annotations: workbenchesAnnotations,
+		},
+		Spec: componentsv1.WorkbenchesSpec{
+			WorkbenchesCommonSpec: dsc.Spec.Components.Workbenches.WorkbenchesCommonSpec,
+		},
+	})
+}
+
+func (s *componentHandler) Init(platform cluster.Platform) error {
+	imageParamMap := map[string]string{
+		"odh-notebook-controller-image":    "RELATED_IMAGE_ODH_NOTEBOOK_CONTROLLER_IMAGE",
+		"odh-kf-notebook-controller-image": "RELATED_IMAGE_ODH_KF_NOTEBOOK_CONTROLLER_IMAGE",
+	}
+
+	// for kf-notebook-controller image
+	if err := odhdeploy.ApplyParams(NotebookControllerPath, imageParamMap); err != nil {
```

Comment:
a nit:
here it only needs "odh-notebook-controller-image": "RELATED_IMAGE_ODH_NOTEBOOK_CONTROLLER_IMAGE",